### PR TITLE
Expose kubevirt-configurations field

### PR DESCRIPTION
**Pull request by alonSadan**
**What this PR does / why we need it**:
We expose a new api-type
which holds all kubevirt configurations. HCO can now vendor this new type
and, and use it to put values in virt-ConfigMap.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
This pr goes hand in hand with https://github.com/kubevirt/hyperconverged-cluster-operator/pull/391.
I will update the pr in HCO accordingly.
**Release note**:

```release-note
Add KubevirtConfigurations, MigrationConfig, DeveloperConfigurations and NetworkConfigurations to API-types
```

### CODE REVIEW

File: pkg/virt-config/config-map.go [hunk 1]
```diff
@@ -44,22 +44,25 @@ import (
...
-	useEmulationKey                   = "debug.useEmulation"
-	ImagePullPolicyKey                = "dev.imagePullPolicy"
-	MigrationsConfigKey               = "migrations"
-	CpuModelKey                       = "default-cpu-model"
-	CpuRequestKey                     = "cpu-request"
-	MemoryOvercommitKey               = "memory-overcommit"
-	LessPVCSpaceTolerationKey         = "pvc-tolerate-less-space-up-to-percent"
-	NodeSelectorsKey                  = "node-selectors"
-	NetworkInterfaceKey               = "default-network-interface"
+	FeatureGatesKey                   = "featureGates"
```
Comment (rmohr) [inline pkg/virt-config/config-map.go]: Maybe I miss something, but this would probably break all existing installations which have setting on kubevirt.

Comment (rmohr) [reply pkg/virt-config/config-map.go]: We need to support both for some time. If no kubevirt yaml is presented in the config map, the config should still be taken as-is from the config map with the old names.

Comment (alonSadan) [reply pkg/virt-config/config-map.go]: Thank you @rmohr :)
I hope now it's OK.

**Comment by alonSadan** [comment]
Hey @rmohr Can you take a second look when you have time?
Thanks :)

**Comment by alonSadan** [comment]
This PR is obsolete since the desired change is already done here: https://github.com/kubevirt/kubevirt/pull/3455

**Pull request closed without merging**